Content.Server/GameObjects/Components/Body/Behavior/BrainBehaviorComponent.cs

```diff
@@ -0,0 +1,91 @@
+﻿#nullable enable
+using Content.Server.GameObjects.Components.Mobs;
+using Content.Server.Mobs;
+using Content.Shared.GameObjects.Components.Body;
+using Content.Shared.GameObjects.Components.Body.Behavior;
+using Content.Shared.GameObjects.Components.Body.Part;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.ComponentDependencies;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Log;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Body.Behavior
+{
+    [RegisterComponent]
+    public class BrainBehaviorComponent : MechanismBehaviorComponent
+    {
+        public override string Name => "Brain";
+
+        protected override void OnAddedToBody()
+        {
+            base.OnAddedToBody();
+
+            if (Body == null)
+                return;
+
+            HandleMind(Body.Owner, Owner);
+        }
+
+        protected override void OnAddedToPart()
+        {
+            base.OnAddedToPart();
+
+            if (Part == null)
+                return;
+
+            HandleMind(Part.Owner, Owner);
+        }
+
+        protected override void OnAddedToPartInBody()
+        {
+            base.OnAddedToPartInBody();
+
+            if (Body == null || Part == null)
+                return;
```

Comment:
Admittedly the IDE is going to complain here but these should never be null either. Probably better to DebugTools.AssertNotNull at most if you really wanted, but that is automatically done for you already.